### PR TITLE
Match file locations with (expected) kernel ones.

### LINKED ISSUE #33: Match file locations with the kernel tree
Could we move files in the repository to the same location as in the kernel repository?

I. e.:
- drivers/hwmon/nzxt-*.c
- Documentation/hwmon/nzxt-*.rst

It should make it much easier to move patches between this repository and the kernel

### PR DESCRIPTION
**Pull request by amezin**
* Move sources to `drivers/hwmon`.

* Move docs to `Documentation/hwmon`.

Closes #33

Signed-off-by: Aleksandr Mezin <mezin.alexander@gmail.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,10 +70,10 @@ built modules can then be loaded with `insmod`.
...
 ```
 $ make
-$ sudo insmod nzxt-grid3.ko             # NZXT Grid+ V3/Smart Device (V1)
-$ sudo insmod nzxt-kraken2.ko           # NZXT Kraken X42/X52/X62/X72
-$ sudo insmod nzxt-kraken3              # NZXT Kraken X53/X63/X73
-$ sudo insmod nzxt-smart2               # NZXT Smart Device V2/RGB & Fan Controller
+$ sudo insmod drivers/hwmon/nzxt-grid3.ko   # NZXT Grid+ V3/Smart Device (V1)
+$ sudo insmod drivers/hwmon/nzxt-kraken2.ko # NZXT Kraken X42/X52/X62/X72
+$ sudo insmod drivers/hwmon/nzxt-kraken3.ko # NZXT Kraken X53/X63/X73
+$ sudo insmod drivers/hwmon/nzxt-smart2.ko  # NZXT Smart Device V2/RGB & Fan Controller
```
Comment (jonasmalacofilho) [inline README.md:76]: The missing `.ko` extensions needed to be fixed, but I don't find having the comments squished together with the commands all that readable.

But it's a minor thing, so feel free to keep it this way... just then please do the same with `modprobe` commands a few paragraphs above.

**Comment by jonasmalacofilho** [comment]
While you're at it, you can also delete `nzxt-smartdevice.rst`. As far as I can see/remember, it's just a mistake that 00bcb937be27caf0ee0a6fb0f6c191474c858e0b didn't remove it.

**Comment by jonasmalacofilho** [comment]
Please rebase to remove duplicate commits (now that #41 has been merged).

**Comment by amezin** [comment]
@jonasmalacofilho Please take a look at [Settings](https://github.com/liquidctl/liquidtux/settings) -> Branches -> Branch protection rules -> Require status checks to pass before merging -> Require branches to be up to date before merging

Also, is anything wrong with "Rebase and merge" method? It keeps the original commit message

**Comment by amezin** [comment]
@jonasmalacofilho So, may I merge this?

> While you're at it, you can also delete nzxt-smartdevice.rst

Should be a separate change/pull request IMO

**Review by jonasmalacofilho — COMMENTED** [review]
Nothing wrong here, but let's not move files from under #44.

---

> Should be a separate change/pull request IMO

Ok

**Comment by amezin** [comment]
> Nothing wrong here, but let's not move files from under #44.

That can be resolved by `git rebase` automatically (but not GitHub UI though). I've already checked by rebasing this PR after https://github.com/liquidctl/liquidtux/commit/e5b8c7165496cf21329d0ff290adf4e5e5e19c2e. File moved in one branch, modified in place in another -> automatic conflict resolution.

Meanwhile this pull request not being merged blocks me from further progress - I have to work with `master` branch, then manually fix this pull request (when the changes affect `Makefile`, for example).

cc @aleksamagicka

**Comment by jonasmalacofilho** [comment]
> Meanwhile this pull request not being merged blocks me from further progress - I have to work with `master` branch, then manually fix this pull request (when the changes affect `Makefile`, for example).

Your argument is that you're annoyed to have to do this, so I should push the burden on the other person?

If that's it, then sorry, no, an open – and nearly ready – PR with driver changes takes priority.

**Comment by stalkerg** [comment]
anyway, this PR should be updated

**Review by jonasmalacofilho — APPROVED** [review]

**Pull request merged**